libnmstate/nm/applier.py

```diff
@@ -170,17 +170,18 @@ def set_ifaces_admin_state(ifaces_desired_state, con_profiles=()):
                 InterfaceState.ABSENT,
             ):
                 nmdevs = _get_affected_devices(iface_desired_state)
-                for nmdev in nmdevs:
-                    remove_devs_actions[nmdev] = [
+                for affected_nmdev in nmdevs:
+                    remove_devs_actions[affected_nmdev] = [
                         device.deactivate,
                         device.delete,
                     ]
-                    if nmdev.get_device_type() in (
-                        nmclient.NM.DeviceType.OVS_BRIDGE,
-                        nmclient.NM.DeviceType.OVS_PORT,
-                        nmclient.NM.DeviceType.OVS_INTERFACE,
-                    ):
-                        remove_devs_actions[nmdev].append(device.delete_device)
+                if (
+                    nmdev.is_software()
+                    and iface_desired_state[Interface.STATE]
+                    == InterfaceState.ABSENT
+                    and nmdev.get_device_type() != nmclient.NM.DeviceType.VETH
+                ):
+                    remove_devs_actions[nmdev].append(device.delete_device)
```

Comment:
Since the change here is meant to work around certain edge cases, another approach could be to change the success callback for deactivate to remove the interface if it is still there and it should have been removed?

Reply:
Good idea. We should try this in a following PR.